compiler/noirc_frontend/src/elaborator/comptime.rs

```diff
@@ -104,7 +107,14 @@ impl<'context> Elaborator<'context> {
         let result = f(&mut elaborator);
         elaborator.check_and_pop_function_context();
 
-        self.errors.append(&mut elaborator.errors);
+        let mut errors = std::mem::take(&mut elaborator.errors);
+        if let Some(reason) = reason {
+            errors = vecmap(errors, |error| {
+                CompilationError::ComptimeError(reason.to_macro_error(error))
+            });
```

Comment:
I was going to comment on allocating a new Vec possibly being unnecessary here although it seems rust may actually optimize this case to reuse the same Vec.

Reply:
Ah, yes, I thought about it but I couldn't find a method to do it. Or, well, there's `iter_mut()` but I imagined it would make the code a bit longer. I'll try to send a small PR tomorrow to do this.

Reply:
It's alright - seems Rust already optimizes this case after all. You won't be able to use `.iter_mut` without cloning the error itself, and I don't think you'll be able to use `drain` either since it'd borrow the vector while it is draining and you wouldn't be able to reassign to the same vector while it is borrowed without collecting to another Vec anyway first.